code/globalincs/version.cpp

```diff
@@ -5,85 +5,122 @@
  * or otherwise commercially exploit the source or things you created based on the 
  * source.
  *
-*/ 
+*/
 
 #include "globalincs/pstypes.h"
 #include "globalincs/version.h"
 
-namespace gameversion
-{
-	bool check_at_least(int major, int minor, int build, int revision)
-	{
-		if (FS_VERSION_MAJOR < major)
-		{
-			return false;
-		}
-		if (FS_VERSION_MAJOR > major)
-		{
-			// Major is greater than the given version => the rest doesn't matter
-			return true;
-		}
-		// major is now equal to our major version
-		
-		if (FS_VERSION_MINOR < minor)
-		{
-			return false;
-		}
-		if (FS_VERSION_MINOR > minor)
-		{
-			// Minor is greater than the given version => the rest doesn't matter
-			return true;
-		}
-		// minor is now equal to our minor version
-		
-		if (FS_VERSION_BUILD < build)
-		{
-			return false;
-		}
-		if (FS_VERSION_BUILD > build)
-		{
-			// build is greater than the given version => the rest doesn't matter
-			return true;
-		}
-		// build is now equal to our build version
-		
-		if (revision == 0)
-		{
-			// Special case, if there is no revision info, skip it
-			return true;
-		}
-		if (FS_VERSION_REVIS == 0)
-		{
-			// Special case, when there is no revision ignore it
-			return true;
-		}
-		
-		if (FS_VERSION_REVIS < revision)
-		{
-			return false;
-		}
-		if (FS_VERSION_REVIS > revision)
-		{
-			// build is greater than the given version => the rest doesn't matter
-			return true;
-		}
-		
-		// revision is now equal to our revision version
+#include "parse/parselo.h"
+
+namespace gameversion {
+
+bool check_at_least(const version& v) {
+	return get_executable_version() < v;
```

Comment:
`check_at_least` is actually correct but `operator<` is literally the exact opposite of "less-of". I'll fix that and write a few tests to make sure it works correctly.